clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -563,21 +574,24 @@ private void reenqueueBatch(ProducerBatch batch, long currentTimeMs) {
     }
 
     private void completeBatch(ProducerBatch batch, ProduceResponse.PartitionResponse response) {
-        if (transactionManager != null && transactionManager.hasProducerIdAndEpoch(batch.producerId(), batch.producerEpoch())) {
-            transactionManager.incrementSequenceNumber(batch.topicPartition, batch.recordCount);
-            log.debug("Incremented sequence number for topic-partition {} to {}", batch.topicPartition,
-                    transactionManager.sequenceNumber(batch.topicPartition));
+        if (transactionManager != null) {
+            if (transactionManager.hasProducerIdAndEpoch(batch.producerId(), batch.producerEpoch())) {
+                transactionManager.maybeUpdateLastAckedSequence(batch.topicPartition, batch.baseSequence() + batch.recordCount - 1);
```

Comment:
Actually, the log message should contain the actual lastAckedSequence since the update is conditional: with out of order responses, it is possible that we might move the sequence backward, so the `maybeUpadteLastAckedSequence` guards against that.

Reply:
Pardon.

In the current code base, I don't see maybeUpadteLastAckedSequence.
What's the method called now ?